kshape/core.py

```diff
@@ -155,34 +157,83 @@ def _extract_shape(idx, x, j, cur_center):
     return zscore(centroid, ddof=1)
 
 
-def _kshape(x, k):
+def _kshape(x, k,n_init=1, max_iter=100, n_jobs = 1, random_state=None):
     """
     >>> from numpy.random import seed; seed(0)
     >>> _kshape(np.array([[1,2,3,4], [0,1,2,3], [-1,1,-1,1], [1,2,2,3]]), 2)
     (array([0, 0, 1, 0]), array([[-1.2244258 , -0.35015476,  0.52411628,  1.05046429],
            [-0.8660254 ,  0.8660254 , -0.8660254 ,  0.8660254 ]]))
     """
-    m = x.shape[0]
-    idx = randint(0, k, size=m)
+    random_state = check_random_state(random_state)
```

Comment:
So this creates a different random clustering every time it runs? Would a fixed seed approach not be better?

Reply:
The partitional clustering algorithm only converges to local solutions. Even though each initialization eventually converges, it converges to a different solution. 

If you fix the seed before calling kshape(), it will run the same clustering every time you run kshape(). See https://github.com/holgerteichgraeber/kshape/blob/master/demo/example_fixedseed.py

Reply:
Yes, but then loop would create different starting states in a deterministic manner, no? Having a fixed seed makes unit testing much easier.

Reply:
I suppose equal distribution is the goal here, not true randomness.